Auto sync upstream 2022 08 04 07 47

nunnatsa opened the pull request:
<!-- Thanks for sending a pull request! -->

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release notes**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
Auto sync upstream 2022-08-04
```

elmiko commented:
this is going to need these 2 patches to merge, then be re-vendored to include them:
https://github.com/openshift/cluster-api-provider-azure/pull/216
https://github.com/openshift/cluster-api-provider-aws/pull/407
/hold

elmiko commented:
this is failing go vet due to an old go lang version (1.13)

elmiko commented:
/retest

elmiko commented:
updated to include a controller-runtime bump, see https://github.com/openshift/cluster-api-provider-baremetal/pull/151#discussion_r631230247

nirarg commented:
/test e2e-kubevirt

nirarg commented:
/lgtm

agradouski committed:
[e2e tests] Cover creating tenant clusters in external infra clusters in the integration tests

Commit:
added integration with the image-builder kubevirt image

k8s-ci-robot committed:
Merge pull request #108 from isaacdorfman/image-builder-integration

Added integration with the image-builder kubevirt image

nirarg committed:
kubevirtci: Add option to interact with the tenant cluster and improve usage (help)

nunnatsa committed:
Add failure printout to the e2e test RunCmd function

In order to debug e2e test failures, add failure printout to the RunCmd
function, so we could understand what went wrong in case of a failure.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

k8s-ci-robot committed:
Merge pull request #134 from nunnatsa/debug-e2e

Add failure printout to the e2e test RunCmd function

Commit:
added docs for creating the image-builder image

Commit:
clusterctl integration

Commit:
Fixed a linting problem

k8s-ci-robot committed:
Merge pull request #113 from isaacdorfman/clusterctl-integration

Clusterctl integration

k8s-ci-robot committed:
Merge pull request #135 from isaacdorfman/image-builder-docs

added docs for creating the image-builder image

nunnatsa committed:
Move to golang 1.18

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

k8s-ci-robot committed:
Merge pull request #136 from nunnatsa/move-to-go-1.18

Move to golang 1.18

nirarg committed:
Add tenant cluster verification to createcluster test

Add the ability to access the tenant cluster during the test
Currently checking only the Nodes exist, need to add more functionality
(For example adding new operators/workloads using the network and storage)
In order to have this ability, added the following:
1. Using virtctl client for kubevirt resources
2. Performing Virtctl portforward and modifying the tenant cluster
   kubeconfig

k8s-ci-robot committed:
Merge pull request #133 from nirarg/kubevirtci-tenant-access

kubevirtci: Add option to interact with the tenant cluster and improve usage (help)

davidvossel committed:
Add namespace to kubectl-tenant command

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Add ssh commands for infra and tenant nodes

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
add kubevirtci destroy-cluster command

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
make separate kubevirtci command for simulating external clusters

Signed-off-by: David Vossel <davidvossel@gmail.com>

k8s-ci-robot committed:
Merge pull request #140 from davidvossel/kubevirtci-updates

Kubevirtci updates

davidvossel committed:
Add kubevirtci option for installing metallb

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Adds command to curl a load balancer within infra cluster

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Add kubevirtci command to install capk from upstream release

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Add kubevirtci function to clean local cache

Signed-off-by: David Vossel <davidvossel@gmail.com>

k8s-ci-robot committed:
Merge pull request #141 from davidvossel/kubevirtci-metal-lb

Kubevirtci metal lb support

k8s-ci-robot committed:
Merge pull request #139 from nirarg/e2e-test

Add tenant cluster verification to create-cluster test

davidvossel committed:
add install calico to kubevirtci and update cluster-up git hash

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
offset tenant network settings from default infra kubeadm settings

The domain and cidr ranges of the tenant cluster need to be offset
from the default kubeadm settings or else they will likely conflict
and overlap with the infra network's settings

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Install cni and verify pod/node status in e2e tests

Signed-off-by: David Vossel <davidvossel@gmail.com>

k8s-ci-robot committed:
Merge pull request #143 from davidvossel/network-fixes

Network fixes

davidvossel committed:
Fix flaky e2e test

The test occasionally failed because the Update() vmi call
returned an err. This err occurred because the vmi would mutate
between the Get and Update calls. To fix this, we use a Patch

Signed-off-by: David Vossel <davidvossel@gmail.com>

k8s-ci-robot committed:
Merge pull request #144 from davidvossel/e2e-test-fix

Fix flaky e2e test

nunnatsa committed:
Graceful deletion of a VirtualMachineInstance

When a host node is deleted, the guest VMIs are evicted, killing the
running tenant node in this VMI.

This PR gracefully delete the VMI, by first drain the guest node. Only
if the drain successful, CAPK will delete the VMI.

The PR is based on a feature in KubeVirt, where by setting the eviction
strategy to "external" KubeVirt will not delete the VMI, but will set
the `vmi.Status.EvacuationNodeName` field to signal to the external
controller (CAPK in this case) that the VMI should be evicted.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
Apply code-review comments

Also, add unit tests. Needed to changed the logic to get the guest
cluster client, to use the workloadClient type, because the previous
code was not testable.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
Drain node: add timeout

If drain is failing for more than 10 minutes, delete the VMI anyway.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
Add VMI eviction functional test

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

davidvossel committed:
Revert "Fix flaky e2e test"

k8s-ci-robot committed:
Merge pull request #148 from davidvossel/revert-144-e2e-test-fix

Revert "Fix flaky e2e test"

pjaton committed:
Enable specifying namespace where VM and VMI are created.

Also, add tests of VM deletion.

pjaton committed:
Search for external VM to delete in correct namespace.

The previous logic always look in the KubeVirtMachine namespace for the
VM to delete instead of using the resolve infrastructure namespace.

qinqon committed:
kubevirtci: Wait for cluster and calico

Some guards are missing from kubevirtci script, this change add them:
- Wait for cluster readiness after creation
- Wait for vms readiness
- Wait for calico node pod readiness
- Wait for tenant node readiness

Signed-off-by: Enrique Llorente <ellorent@redhat.com>

davidvossel committed:
Pin kubevirtci to a kubevirt release to prevent CI regressions

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Update to k8s 1.22 guest cluster image

Signed-off-by: David Vossel <davidvossel@gmail.com>

pjaton committed:
Avoid reporting not-found errors when patching resources under deletion.

It happens that the resources are already deleted by the time the
controller tries to patch them during reconcile-delete. In this case,
the code should be lenient and not report on not-found errors.

davidvossel committed:
Reduce memory usage by using local storage provider

Signed-off-by: David Vossel <davidvossel@gmail.com>

k8s-ci-robot committed:
Merge pull request #155 from davidvossel/pin-kubevirt-release

Unblock e2e CI tests

pjaton committed:
Add capk user properly when users already defined.

This change takes into account the case where the user configured
additional users already, including the case when a capk user is already
added.

pjaton committed:
Select kubeconfig current context namespace before 'default'.

Update the logic resolving the infrastructure namespace to use the
namespace of the current context, if any, in the kubeconfig before
defaulting to the 'default' namespace.

k8s-ci-robot committed:
Merge pull request #151 from pjaton/fix-external-deletion

Fix external deletion

k8s-ci-robot committed:
Merge pull request #132 from agradouski/issue/90

[e2e tests] Cover creating tenant clusters in external infra clusters in the integration tests

k8s-ci-robot committed:
Merge pull request #157 from pjaton/fix-patch-of-deleted-failure

Avoid reporting not-found errors when patching resources under deletion.

k8s-ci-robot committed:
Merge pull request #160 from pjaton/fix-namespace-logic

Select kubeconfig current context namespace before 'default'.

pjaton committed:
Copy Kubeadm userdata secret labels to CAPK secret.

k8s-ci-robot committed:
Merge pull request #154 from qinqon/wait-for-tenant

kubevirtci: Wait for cluster and calico

pjaton committed:
Use gomega matcher to check labels

nunnatsa committed:
fix bug in the vmi eviction controller - wrong predicate

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

k8s-ci-robot committed:
Merge pull request #159 from pjaton/fix-capk-user-add

Add capk user properly when users already defined.

pjaton committed:
Refactor setup-scheme into shared functions.

pjaton committed:
Look for the infrastructure secret in the owner namespace by default.

Also, add unit-test for the infracluster package.

k8s-ci-robot committed:
Merge pull request #164 from pjaton/add-secret-labels

Copy Kubeadm userdata secret labels to CAPK secret.

k8s-ci-robot committed:
Merge pull request #167 from pjaton/fix-infra-secret-lookup

Fix infra secret lookup so that it defaults to the namespace of the referencing resource

nunnatsa committed:
Fix code review comments

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

k8s-ci-robot committed:
Merge pull request #145 from nunnatsa/drain-node

Graceful deletion of a VirtualMachineInstance

nunnatsa committed:
Merge remote-tracking branch 'auto-sync-script-upstream/main' into auto_sync_upstream_2022-08-04-07-47

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

davidvossel commented:
/retest

davidvossel commented:
ci is failing

```
+ sleep 60
+ oc wait pod --for=condition=Ready -l cluster.x-k8s.io/provider=infrastructure-kubevirt -n capk-system --timeout=120s
error: no matching resources found
```

nunnatsa commented:
too old
/close
